README.md

```diff
@@ -114,3 +115,151 @@ The file should be named according to your Operating System.
+   the `target/org.wso2.carbon.securevault.hashicorp-1.0.jar` file to `<IS_HOME>/repository/components/dropin/`
+   directory.
+
+2. Add **HashiCorp Vault Java Driver** (Eg: `vault-java-driver-5.1.0.jar`) to the
+   `<IS_HOME>/repository/components/lib/` directory.
+
+3. Open `/repository/conf/security/secret-conf.properties` file and set following configurations.
+    ```
+    keystore.identity.location=repository/resources/security/wso2carbon.jks
+    keystore.identity.type=JKS
+    keystore.identity.store.password=identity.store.password
+    keystore.identity.store.secretProvider=org.wso2.carbon.securevault.DefaultSecretCallbackHandler
+    keystore.identity.key.password=identity.key.password
+    keystore.identity.key.secretProvider=org.wso2.carbon.securevault.DefaultSecretCallbackHandler
+    carbon.secretProvider=org.wso2.securevault.secret.handler.SecretManagerSecretCallbackHandler
+    
+   secVault.enabled=true
+   secretRepositories=vault
+   secretRepositories.vault.provider=org.wso2.carbon.securevault.hashicorp.repository.HashiCorpSecretRepositoryProvider
+   secretRepositories.vault.properties.address=https://127.0.0.1:8200
+   secretRepositories.vault.properties.namespace=ns1
+   secretRepositories.vault.properties.enginePath=wso2is
+   secretRepositories.vault.properties.engineVersion=2
+   secretRepositories.vault.properties.authType=APP_ROLE
+   secretRepositories.vault.properties.roleId=<role id>
+    ```
+
+   **Note:** In production, you should always use the vault address with TLS enabled.
+
+4. Add following lines to the `<IS_HOME>/repository/conf/log4j2.properties` file
+    ```
+    logger.org-wso2-carbon-securevault-hashicorp.name=org.wso2.carbon.securevault.hashicorp
+    logger.org-wso2-carbon-securevault-hashicorp.level=INFO
+    logger.org-wso2-carbon-securevault-hashicorp.additivity=false
+    logger.org-wso2-carbon-securevault-hashicorp.appenderRef.CARBON_CONSOLE.ref = CARBON_CONSOLE
+    ```
+   Then append `org-wso2-carbon-securevault-hashicorp` to the `loggers` list in the same file as follows.
+   ```
+   loggers = AUDIT_LOG, trace-messages, ... ,org-wso2-carbon-securevault-hashicorp
+   ```
+
+### Step 3: Update passwords with their aliases
+1. Open the `deployment.toml` file in the `<IS_HOME>/repository/conf/` directory and add
+   the `[secrets]` configuration section **at the bottom of the file** as shown below.
+   Give an alias for the passwords and put the value as blank (`""`).
+
+    ```toml
+    [secrets]
+    admin_password = ""
+    ```
+
+2. Add the encrypted password alias to the relevant sections in the `deployment.toml`
+   file by using a place holder: `$secret{alias}`. For example:
+
+    ```toml
+    [super_admin]
+    username="admin"
+    password="$secret{admin_password}"
+    ```
+
+### Step 4: Start the Server
+
+1. Provide the `VAULT_TOKEN` to the prompted message in the console or by creating a new file in the `<IS_HOME>`
```

Comment:
vault token or secret id?

Reply:
corrected in 4f7d88b2f3af285a2c77dffcb2594d8d586d2e1c